apps/gradient/gradient.star

```diff
@@ -0,0 +1,295 @@
+Author: Jeffrey Lancaster
+"""
+
+load("math.star", "math")
+load("random.star", "random")
+load("render.star", "render")
+load("schema.star", "schema")
+
+PIXLET_W = 64
+PIXLET_H = 32
+
+def median(val1, val2):
+    return math.floor((val1 + val2) / 2)
+
+def makeRange(minValue, maxValue, numValues):
+    rangeArray = []
+    for i in range(0, numValues):
+        step = (maxValue - minValue) / numValues
+        calcValue = math.round(minValue + (i * step))
+        rangeArray.append(calcValue)
+    return rangeArray
+
+def rgbRange(start, end, steps):
+    rRange = makeRange(start[0], end[0], steps)
+    gRange = makeRange(start[1], end[1], steps)
+    bRange = makeRange(start[2], end[2], steps)
+    returnRange = []
+    for n in range(0, steps):
+        returnRange.append([rRange[n], gRange[n], bRange[n]])
+    return returnRange
+
+# from: https://www.educative.io/answers/how-to-convert-hex-to-rgb-and-rgb-to-hex-in-python
+def hex_to_rgb(hex):
+    hex = hex.replace("#", "")
+    rgb = []
+    for i in (0, 2, 4):
+        decimal = int(hex[i:i + 2], 16)
+        rgb.append(decimal)
+    return tuple(rgb)
+
+def rgb_to_hex(r, g, b):
+    rgbArr = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "A", "B", "C", "D", "E", "F"]
+    hex = "#"
+    r = math.floor(r)
+    g = math.floor(g)
+    b = math.floor(b)
+    for i in (r, g, b):
+        secondNum = i % 16
+        firstNum = math.floor((i - secondNum) / 16)
+        hex += rgbArr[firstNum] + rgbArr[secondNum]
+    return hex
+
+def randomColor():
+    randomRed = random.number(0, 255)
+    randomGreen = random.number(0, 255)
+    randomBlue = random.number(0, 255)
+    return rgb_to_hex(randomRed, randomGreen, randomBlue)
+
+def four_color_gradient(topL, topR, botL, botR):
+    # convert inputs to rgb
+    topLrgb = hex_to_rgb(topL)
+    topRrgb = hex_to_rgb(topR)
+    botLrgb = hex_to_rgb(botL)
+    botRrgb = hex_to_rgb(botR)
+
+    # determine left column and right column ranges: PIXLET_H
+    leftCol = rgbRange(topLrgb, botLrgb, PIXLET_H)
+    rightCol = rgbRange(topRrgb, botRrgb, PIXLET_H)
+
+    # for each row, determine range: PIXLET_W
+    gradientArray = []
+    for n in range(0, PIXLET_H):
+        rowGradient = rgbRange(leftCol[n], rightCol[n], PIXLET_W)
+
+        # convert row to hex values
+        for n, i in enumerate(rowGradient):
+            rowGradient[n] = rgb_to_hex(i[0], i[1], i[2])
+        gradientArray.append(rowGradient)
+
+    return gradientArray
+
+def two_color_gradient(topL, botR):
+    topLrgb = hex_to_rgb(topL)
+    botRrgb = hex_to_rgb(botR)
+    medianR = median(topLrgb[0], botRrgb[0])
+    medianG = median(topLrgb[1], botRrgb[1])
+    medianB = median(topLrgb[2], botRrgb[2])
+
+    # average r, g, b for other two corners
+    medianRGB = rgb_to_hex(medianR, medianG, medianB)
+    return four_color_gradient(topL, medianRGB, medianRGB, botR)
+
+def main(config):
+    # define gradientArray and labels
```

Comment:
```suggestion
    random.seed(time.now().unix // 15)

    # define gradientArray and labels
```

Reply:
This will make the app a little bit kinder on our internal caches of rendered graphics. Keeps the randomness deterministic over 15 second time windows.